pdu/UDH.go

```diff
@@ -15,27 +17,35 @@ import (
 type UDH []InfoElement
 
 // UDHL return the length (number of octet) of the encoded UDH itself
-func (u UDH) UDHL() (l int) {
-	for i := range u {
-		l += len(u[i].Data)
+func (u *UDH) UDHL() int {
+	if u == nil {
+		return 0
 	}
-	return l
+	length := 1
+	for _, element := range *u {
```

Comment:
Copy instead of referencing is no need. Be aware of range loop with value.

Reply:
this is array.

```go
u  // type of `*[]InfoElement`
*u // type of `[]InfoElement` 
```

Reply:
@septs Because you update function signature to use pointer right? What is the purpose of this?

And what I concern is `_, element := range`. Please be aware that `element` is stack-cloned of InfoElement. This copy ops is no need! In this case, please use index loop: `for index := range`